openpype/modules/deadline/plugins/publish/submit_maya_deadline.py

```diff
@@ -440,7 +441,8 @@ def process(self, instance):
 
         output_filename_0 = filename_0
 
-        dirname = os.path.dirname(output_filename_0)
+        if render_variables["renderer"] == "renderman":
+            dirname = os.path.dirname(output_filename_0)
```

Comment:
Should we add a comment as to why this workaround is needed for renderman?

Reply:
Also might  be good to be aware that elsewhere in the code `renderer` is taken from `instance.data` instead of re-read from the layer: https://github.com/pypeclub/OpenPype/blob/3744b21266258fdd69814807a4152e268dca3d27/openpype/modules/deadline/plugins/publish/submit_maya_deadline.py#L731

As you said, cleanup of this code is needed. :)

Reply:
you are right, I've added those changes